source/memory/memory.go

```diff
@@ -30,6 +31,24 @@ type Source struct{}
 // Name returns an identifier string for this feature source.
 func (s Source) Name() string { return "memory" }
 
+var consecutiveSpaces = regexp.MustCompile("[[:space:]]+")
+
+func GetChoice(parameter string) (string, error) {
```

Comment:
Add a docstring to the function. Also, arg name 'parameter' is misleading/vague, IMO. Could you use e.g. 'path' or smth?

Reply:
Done.